src/main/java/org/jenkinsci/test/acceptance/junit/WithOS.java

```diff
@@ -0,0 +1,89 @@
+package org.jenkinsci.test.acceptance.junit;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.util.Arrays;
+
+import org.apache.commons.lang.SystemUtils;
+import org.junit.internal.AssumptionViolatedException;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * 
+ * Indicates that the test and Jenkins instance must be running 
+ * on one of the operating systems provided. If this condition is not
+ * met, this test will be skipped.
+ * 
+ * <p>
+ * This check can only be <strong>trusted to do its job correctly</strong> when tests 
+ * are being run with a LocalController. When using a different controller, 
+ * the machine running Jenkins can be different from the one running the tests so
+ * the test may pass or fail when it should not.
```

Comment:
Yet you are not actually issuing a runtime warning when this is the case, which you probably [could](https://github.com/jenkinsci/acceptance-test-harness/blob/6681f400d9a84ded4c39e8a8c4c982c1a2196295/src/main/java/org/jenkinsci/test/acceptance/junit/JenkinsAcceptanceTestRule.java#L169) [be](https://github.com/jenkinsci/acceptance-test-harness/blob/6681f400d9a84ded4c39e8a8c4c982c1a2196295/src/main/java/org/jenkinsci/test/acceptance/junit/WithPlugins.java#L73-L77).